### PR TITLE
Show error rate if Indexing some of the documents failed

### LINKED ISSUE #1509: Ingest should show errors at the end
The last line looks like a success

```
 Num docs      24 Parse errs    24 PublSplits   1 Input size     0MB Thrghput  0.00MB/s Time 00:00:03
 Num docs      24 Parse errs    24 PublSplits   1 Input size     0MB Thrghput  0.00MB/s Time 00:00:03

Indexed 24 documents in 3s
```

### PR DESCRIPTION
**Pull request by boraarslan**
Closes #1509 
<sup>_I think_</sup>

### CODE REVIEW

File: quickwit-cli/src/index.rs [hunk 1]
```diff
@@ -1088,6 +1088,15 @@ pub async fn start_statistics_reporting_loop(
         pipeline_statistics.num_docs.separate_with_commas(),
         format_duration(secs)
     );
+    if pipeline_statistics.num_invalid_docs > 0 {
```
Comment (guilload) [inline quickwit-cli/src/index.rs:1091]: Let's just display one line:
```rust
if pipeline_statistics.num_invalid_docs == 0 {
    // logic unchanged
    Ok(pipeline_statistics)
} else {
    println!("Indexed {num_indexed_docs/num_docs} documents in 3s ({:.1}% success rate).")
    Err()
}
```

File: quickwit-cli/src/index.rs [hunk 2]
```diff
@@ -1088,6 +1088,15 @@ pub async fn start_statistics_reporting_loop(
...
+        println!(
+            "Failed to index {} documents. ({:.1}% success rate)",
+            pipeline_statistics.num_invalid_docs.separate_with_commas(),
+            1.0 - (pipeline_statistics.num_invalid_docs as f64
+                / pipeline_statistics.num_docs as f64)
+                * 100.0
+        );
+    }
 
     Ok(pipeline_statistics)
```
Comment (guilload) [inline quickwit-cli/src/index.rs:1101]: Let's also make sure the CLI command does not return 0 where there were some errors.

**Review by guilload — CHANGES_REQUESTED** [review]

File: quickwit-cli/src/index.rs (hunk 2 above)
Comment (boraarslan) [reply quickwit-cli/src/index.rs:1112]: The quickest workaround was to check the number of invalid docs on the caller function. Returning error from the report loop might cause skipping clearing the cache.

File: quickwit-cli/src/index.rs [hunk 3]
```diff
@@ -1083,18 +1086,26 @@ pub async fn start_statistics_reporting_loop(
...
     let secs = Duration::from_secs(start_time.elapsed().as_secs());
-    println!(
-        "Indexed {} documents in {}",
-        pipeline_statistics.num_docs.separate_with_commas(),
-        format_duration(secs)
-    );
-    if pipeline_statistics.num_invalid_docs > 0 {
+    if pipeline_statistics.num_invalid_docs == 0 {
+        println!(
+            "Indexed {} documents in {}",
```
Comment (guilload) [inline quickwit-cli/src/index.rs:1091]:
```suggestion
            "Indexed {} documents in {}.",
```

File: quickwit-cli/src/index.rs [hunk 4]
```diff
@@ -872,7 +872,10 @@ pub async fn ingest_docs_cli(args: IngestDocsArgs) -> anyhow::Result<()> {
         .await?;
     }
 
-    Ok(())
+    match statistics.num_invalid_docs {
+        0 => Ok(()),
+        _ => bail!("Failed to ingest all of the documents."),
```
Comment (guilload) [inline quickwit-cli/src/index.rs:877]:
```suggestion
        _ => bail!("Failed to ingest all the documents."),
```

**Review by guilload — APPROVED** [review]

**Pull request merged**